### ISSUE TITLE
Delay scoreboard setting / team joining by a tick on arena join when world changing

### ISSUE THREAD

**Issue by mibby** [Eredrim/pvparena#85]
PVPArena compiled as of commit https://github.com/Eredrim/pvparena/commit/7edee80eb85e0608b3d50e5394112e3e936a77b3
TAB_Premium_v2.8.11-pre4 (Also occurs with non-premium version)
Paper dev 469 (Spigot 1.16.5)

Running into a slight scoreboard conflict with another plugin. I use the plugin TAB to handle all nametags above player heads. It has a configuration to disable itself in certain worlds. I have it disable itself in the world where PVPArena exists since PA handles scoreboards / name recoloring for team joining. However TAB seems to complain a tick before it disables itself since PA sets the scoreboard early.

```
16.02.2021 - 12:06:30 - [TAB v2.8.11-pre4 Premium] Something just tried to add player Jasmil into team free
16.02.2021 - 12:06:32 - [TAB v2.8.11-pre4 Premium] Something just tried to add player Jasmil into team free
16.02.2021 - 12:06:42 - [TAB v2.8.11-pre4 Premium] Something just tried to add player Jasmil into team free
16.02.2021 - 12:06:45 - [TAB v2.8.11-pre4 Premium] Something just tried to add player Jasmil into team free
16.02.2021 - 12:06:45 - [TAB v2.8.11-pre4 Premium] Something just tried to add player Jasmil into team free
16.02.2021 - 12:06:45 - [TAB v2.8.11-pre4 Premium] Something just tried to add player Jasmil into team free
```

Asking the TAB developer, he said PVPArena would need to delay setting a tick. Would it be possible to slightly delay the scoreboard setting or add an option to adjust the setting delay?

![link](https://i.imgur.com/Mo5q29Q.png)

**Comment by Eredrim** [comment]
Hello Mibby,

Current version of PVPArena doesn't correctly support cross-world features (like joining an arena from another world). This lack will normally be fixed in the next major version (currently in heavy development).
PVPArena teleportation functions massively use delaying in the current implementation and there is a risk to generate side effects if I try to add this new delaying.

**Comment by mibby** [comment]
> Current version of PVPArena doesn't correctly support cross-world features (like joining an arena from another world). This lack will normally be fixed in the next major version (currently in heavy development).

But it does work fine for the most part and has worked fine for me for years. Arena setup supports defining the world for locations / region and teleport handling & inventory restoring is fine. This is just a recent incompatibility I've come across.

**Comment by Eredrim** [comment]
May be fixed in 2.0 dev preview. I suggest to give it a new try.
